tablet_qt/core/networkmanager.cpp

```diff
@@ -878,30 +878,65 @@ void NetworkManager::updateTaskSchedules()
     statusMessage(tr("Getting task schedules from") + " " +
                   serverUrlDisplayString());
 
-    serverPost(dict, &NetworkManager::receivedTaskSchedules);
+    serverPost(dict, &NetworkManager::receivedTaskSchedulesAndPatientDetails);
 }
 
-void NetworkManager::receivedTaskSchedules(QNetworkReply* reply)
+void NetworkManager::receivedTaskSchedulesAndPatientDetails(QNetworkReply* reply)
 {
     if (!processServerReply(reply)) {
         return;
     }
 
-    storeTaskSchedules();
+    storeTaskSchedulesAndPatientDetails();
     succeed();
 }
 
 
-void NetworkManager::storeTaskSchedules()
+void NetworkManager::storeTaskSchedulesAndPatientDetails()
 {
     statusMessage(tr("... received task schedules"));
 
     QJsonParseError error;
-    QJsonDocument doc = QJsonDocument::fromJson(
-        m_reply_dict[KEY_TASK_SCHEDULES].toUtf8(), &error
+
+    // ------------------------------------------------------------------------
+    // Patient
+    // ------------------------------------------------------------------------
+    // Note: Unlike in createSinglePatient(), our patient object already
+    // exists. We're just checking that the details match (in case there's been
+    // a change on the server).
+    const QJsonDocument patient_doc = QJsonDocument::fromJson(
+        m_reply_dict[KEY_PATIENT_INFO].toUtf8(), &error
     );
+    if (patient_doc.isNull()) {
+        const QString message = tr("Failed to parse patient info: %1").arg(
+            error.errorString()
+        );
+        statusMessage(message);
+        fail(ErrorCode::JsonParseError, message);
+        return;
+    }
+    const QJsonArray patients_json_array = patient_doc.array();
+    const QJsonObject patient_json = patients_json_array.first().toObject();
+    Patient* patient = m_app.selectedPatient();
+    if (patient) {
+        patient->setPatientDetailsFromJson(patient_json);
```

Comment:
This now means that the "needs upload" icon now never disappears! Is that OK or should we not trigger "needs upload" here?

Reply:
Ah, yes, that is confusing. Agree; we should suppress the "needs upload" icon if this happens and there are no tasks present, I suppose (it's not that data hasn't changed, but we/the users wouldn't care without any tasks). If there are tasks present on the client afterwards, though, things have changed and do need uploading (is that right?).

Reply:
I've suppressed the icon when updating the task schedules.

I realise we can't really update the task schedules before upload because it will mess up the complete status of the tasks yet to be uploaded. We could create a separate API call to update a patient's details and do this prior to upload but still fetch the task schedule updates after upload